jf_agent/jf_jira/jira_download.py

```diff
@@ -713,3 +713,200 @@ def _search_users(
             includeActive=include_active,
         )
     ]
+
+
+@diagnostics.capture_timing()
+@agent_logging.log_entry_exit(logger)
+def download_missing_repos_found_by_jira(issues_to_scan, config, jira_connection, git_connection):
+    print('downloading git repos found by jira development field... ', end='', flush=True)
+
+    missing_repositories = {}
+
+    for issue_id, instance_types in tqdm(
+        issues_to_scan.items(), desc='scanning jira for git repos'
+    ):
+        for instance_type in instance_types:
+            try:
+                repositories = _scan_jira_issue_for_repo_data(
+                    jira_connection, issue_id, instance_type
+                )
+            except JIRAError as e:
+                if e.status_code == 403:
+                    agent_logging.log_and_print(
+                        logger,
+                        logger.ERROR,
+                        f"you do not have the required 'development field' permissions in jira required to scan for missing repos",
+                    )
+                    return []
+
+            for repo in repositories:
+                repo_name = repo['name']
+                repo_url = repo['url']
+
+                if repo_name not in missing_repositories:
+                    missing_repositories[repo_name] = {
+                        'name': repo_name,
+                        'url': repo_url,
+                        'instance_type': instance_type,
+                    }
+
+    print()
+
+    try:
+        _remove_mismatched_repos(
```

Comment:
I know what this does, but a comment here about the what the side effect of this function is might be useful?